pkg/resource/unstructured/composite/composite.go

```diff
@@ -147,7 +147,17 @@ func (c *Unstructured) GetClaimReference() *corev1.ObjectReference {
 
 // SetClaimReference of this Composite resource.
 func (c *Unstructured) SetClaimReference(ref *corev1.ObjectReference) {
-	_ = fieldpath.Pave(c.Object).SetValue("spec.claimRef", ref)
+	// spec.claimRef schema is just subset of ObjectReference
+	// the passed ref might have more fields set, e.g. UID
+	// and they need to be removed, otherwise K8s API server complains about non-existing field
+	// when client sends patch request
+	claimRef := &corev1.ObjectReference{
+		APIVersion: ref.APIVersion,
+		Kind:       ref.Kind,
+		Namespace:  ref.Namespace,
+		Name:       ref.Name,
+	}
+	_ = fieldpath.Pave(c.Object).SetValue("spec.claimRef", claimRef)
```

Comment:
lgtm.

Just curious, why do we use a type if it is not supported?

Reply:
I also wonder if this method needs to live in crossplane-runtime at all, because it is only consumed in crossplane?

Reply:
That's a different question. What @negz and I mean is that `corev1.ObjectReference` is the wrong type. We should have a `ClaimReference` type instead that only has the fields that are allowed to be set.

Reply:
can you change this, i.e. introduce that type?

Reply:
Does anything actually write (or read) the UID field in a claimRef today? If no, I feel fine making the change since while technically breaking at the API level it would functionally be a no-op.